lib/rspec/mocks/message_expectation.rb

```diff
@@ -556,6 +561,7 @@ def invoke_incrementing_actual_calls_by(increment, allowed_to_fail, parent_stub,
           args.unshift(orig_object) if yield_receiver_to_implementation_block?
 
           if negative? || (allowed_to_fail && (@exactly || @at_most) && (@actual_received_count == @expected_received_count))
+            @failed = true
```

Comment:
I want to remove this, and replace it with an `@invoking` check further down.

e.g
```
@invoking = true
if implementation.present?
  implementation.call(*args, &block)
elsif parent_stub
  parent_stub.invoke(nil, *args, &block)
end
@invoking = false
```

Reply:
I think thats safer and I think theres a risk of false positives as is, as it won't record further invocations?